### PR TITLE
Typescript compilation error with latest apollo package

### PR DESCRIPTION
**Pull request by iharmanpannu**
### Fixed typescript compilation error

```sh
ERROR in /Users/harmandeeppannu/Desktop/app/tailwind-ts-app/node_modules/vue-apollo/types/options.d.ts
47:18 Interface 'VueApolloMutationOptions<V, R>' incorrectly extends interface 'MutationOptions<R, OperationVariables>'.
  Types of property 'optimisticResponse' are incompatible.
    Type 'Object | ((this: ApolloVueThisType<V>) => any) | undefined' is not assignable to type 'R | ((vars: OperationVariables) => R) | undefined'.
      Type 'Object' is not assignable to type 'R | ((vars: OperationVariables) => R) | undefined'.
        The 'Object' type is assignable to very few other types. Did you mean to use the 'any' type instead?
          Type 'Object' is not assignable to type 'R'.
            The 'Object' type is assignable to very few other types. Did you mean to use the 'any' type instead?
    45 | }
    46 |
  > 47 | export interface VueApolloMutationOptions<V, R> extends MutationOptions<R> {
       |                  ^
    48 |   mutation: DocumentNode;
    49 |   variables?: VariableFn<V>;
    50 |   optimisticResponse?: ((this: ApolloVueThisType<V>) => any) | Object;
```

### CODE REVIEW

**Comment by jvbianchi** [comment]
related with #577

**Comment by chanlito** [comment]
Please merge this 😬

**Comment by renestalder** [comment]
Tested @iharmanpannu's patch in an application where this was breaking and with the patch it works again. <3.

**Comment by stephen-bunn** [comment]
Is there any reason why this hasn't been merged yet? ❤️

**Comment by redjoker011** [comment]
> Is there any reason why this hasn't been merged yet? ❤️

Me too waiting for this PR to be merged

**Comment by rille111** [comment]
Merge and relase this please! This is a breaking thing.

**Comment by ScreamZ** [comment]
@Akryum Can u take this in consideration ?

**Comment by anymost** [comment]
wish merge this pull request as soon as possible. can't wait to fix this bug.

**Comment by aaronyoung0626** [comment]
Hope this would be merged ASAP, thx a lot!

**Comment by idrislab** [comment]
@Akryum please merge and release this.

File: types/options.d.ts [hunk 1]
```diff
@@ -47,7 +47,7 @@ export interface VueApolloQueryOptions<V, R> extends ExtendableVueApolloQueryOpt
 export interface VueApolloMutationOptions<V, R> extends MutationOptions<R> {
   mutation: DocumentNode;
   variables?: VariableFn<V>;
-  optimisticResponse?: ((this: ApolloVueThisType<V>) => any) | Object;
+  optimisticResponse?: ((this: ApolloVueThisType<V>) => any) | any
```
Comment (airgram) [inline types/options.d.ts]: Here is much better to use strong types: 
```
optimisticResponse?: ((this: ApolloVueThisType<V>) => R) | R
```

**Comment by Akryum** [comment]
Thanks :+1:

**Comment by exls** [comment]
Tooooo late. switched to another tech...

**Comment by ralph** [comment]
`No type errors found` - Finally! 🎉🙌

**Pull request merged**